### ISSUE TITLE
the percentage value of battery is Wrong [greater than 100%]

### ISSUE THREAD

**Issue by yunfan** [i3/i3status#136]
i am running i3status on my chromebook flip's 
the cpu is armv7 type [rockchip rk3288]
the os is archlinux arm
the i3status version is 2.10 [2016-01-01]

and i have grab a screenshot. check it
https://clbin.com/See1Um.png

**Comment by stapelberg** [comment]
Can you please provide your i3status config file and an strace log of i3status running for a couple of seconds (use `strace -s 2048 -v -tt -o /tmp/strace.log i3status`? Also, what’s the output of `acpi`?

**Comment by yunfan** [comment]
hey i am sorry to reply you so late. 

i3status.conf

```
# i3status configuration file.
# see "man i3status" for documentation.

# It is important that this file is edited as UTF-8.
# The following line should contain a sharp s:
# ß
# If the above line is not correctly displayed, fix your editor first!

general {
        colors = true
        interval = 5
}

order += "disk /"
order += "wireless _first_"
order += "battery 0"
order += "load"
order += "tztime local"

wireless _first_ {
        format_up = "W: (%quality at %essid) %ip"
        format_down = "W: down"
}

ethernet _first_ {
        # if you use %speed, i3status requires root privileges
        format_up = "E: %ip"
        format_down = "E: down"
}

battery 0 {
        format = "%status %percentage"
	path = "/sys/class/power_supply/bq27500-0/uevent"
	low_threshold = 10
}

run_watch DHCP {
        pidfile = "/var/run/dhclient*.pid"
}

run_watch VPN {
        pidfile = "/var/run/vpnc/pid"
}

tztime local {
        format = "%Y-%m-%d %H:%M:%S"
}

load {
        format = "%1min"
}

disk "/" {
        format = "%avail"
}
```

acpi output

```
Battery 0: Discharging, 98%, discharging at zero rate - will never fully discharge.
```

strace.log

its too large, about 700k, so please check here https://clbin.com/XeHbJ

**Comment by stapelberg** [comment]
I think the issue is that the power supply in question provides both `POWER_SUPPLY_CHARGE_NOW` (and the corresponding `POWER_SUPPLY_CHARGE_FULL_DESIGN`) and `POWER_SUPPLY_ENERGY_NOW` (without any corresponding values).

i3status’s current code doesn’t handle this case. I think the best way to fix it is to first collect all keys/values and then enter the correct code branch depending on what’s available (as opposed to the current code which stores information as it goes along).

I think this might be a good opportunity to add a test suite to i3status, so that we can ensure different batteries are supported correctly as we modify the battery-related code.

In case anyone wants to volunteer, please feel free.

**Comment by yunfan** [comment]
actually i found that if i could just read the capactity value to print. that would be fine
so can you add a new template value key for battery? like %capacity ?

**Comment by stapelberg** [comment]
Sorry, I don’t understand. Can you rephrase/elaborate what you mean please?

**Comment by yunfan** [comment]
@stapelberg  i mean in `battery 0`'s configure section, you have some variables like  %status, and %percentage. how about to add a new variable named %capacity? i can get that value from /sys/class/power_supply/{$my_battery}/capacity . so i think it wont be too hard to get in c code

if you add this variable, i can temporily just use this in my i3status before you collected all the batteries features and fix this bug

**Comment by stapelberg** [comment]
Feel free to implement that locally, but I’m not keen on adding such a workaround to i3status itself.

**Comment by sinni800** [comment]
For me it actually uses the design capacity rather than the last full capacity for some reason.

This is my uevent file for the battery, the config file is just default.

POWER_SUPPLY_NAME=BAT0
POWER_SUPPLY_STATUS=Discharging
POWER_SUPPLY_PRESENT=1
POWER_SUPPLY_TECHNOLOGY=Li-poly
POWER_SUPPLY_CYCLE_COUNT=0
POWER_SUPPLY_VOLTAGE_MIN_DESIGN=11400000
POWER_SUPPLY_VOLTAGE_NOW=11874000
POWER_SUPPLY_POWER_NOW=6328000
POWER_SUPPLY_ENERGY_FULL_DESIGN=52550000
POWER_SUPPLY_ENERGY_FULL=35190000
POWER_SUPPLY_ENERGY_NOW=26330000
POWER_SUPPLY_CAPACITY=74
POWER_SUPPLY_CAPACITY_LEVEL=Normal
POWER_SUPPLY_MODEL_NAME=00HW020
POWER_SUPPLY_MANUFACTURER=LGC
POWER_SUPPLY_SERIAL_NUMBER=  398

**Comment by nalyDylan** [comment]
I have the same issue, if i do the math it seems like i3status is using `POWER_SUPPLY_ENERGY_FULL_DESIGN` instead of `POWER_SUPPLY_ENERGY_FULL`

uevent (currently displaying ~106%):
```POWER_SUPPLY_NAME=BAT1
POWER_SUPPLY_STATUS=Discharging
POWER_SUPPLY_PRESENT=1
POWER_SUPPLY_TECHNOLOGY=Li-ion
POWER_SUPPLY_CYCLE_COUNT=0
POWER_SUPPLY_VOLTAGE_MIN_DESIGN=7400000
POWER_SUPPLY_VOLTAGE_NOW=7668000
POWER_SUPPLY_POWER_NOW=7529000
POWER_SUPPLY_ENERGY_FULL_DESIGN=43960000
POWER_SUPPLY_ENERGY_FULL=48840000
POWER_SUPPLY_ENERGY_NOW=47150000
POWER_SUPPLY_CAPACITY=96
POWER_SUPPLY_CAPACITY_LEVEL=Normal
POWER_SUPPLY_MODEL_NAME=PABAS0241231
POWER_SUPPLY_MANUFACTURER=LENOVO
POWER_SUPPLY_SERIAL_NUMBER=41167
```

**Comment by stapelberg** [comment]
@nalyDylan Please read https://i3wm.org/i3status/manpage.html#_battery, then set `last_full_capacity = true`.

**Comment by nalyDylan** [comment]
You're right, I don't know why when I was reading through the documentation I didn't think that option would do exactly what it does. Thanks!
